core/base/src/TDirectory.cxx

```diff
@@ -1068,7 +1068,6 @@ TDirectory *TDirectory::mkdir(const char *name, const char *title, Bool_t return
          tmpdir = mkdir(workname,title);
       delete[] workname;
       if (!tmpdir) return nullptr;
-      if (!newdir) newdir = tmpdir;
```

Comment:
Should the declaration of `newdir` be removed, too, and the function's last two lines be changed to
```
return new TDirectory(name, title, "", this);
```